cmd/app/run.go

```diff
@@ -22,38 +21,35 @@ func Run() {
 	results := message.Results{Map: make(map[string]bool)}
 	failedCount := 0
 
-	regex, err := regexp.Compile("^(http|https)://")
-	if err != nil {
-		log.Println("Failed to create new regex object")
-		return
-	}
+	socketChan := make(chan socket.Result)
+	var wg sync.WaitGroup
 
-	// iterate over given/loaded sockets
+	// iterate over given/loaded sockets --> start goroutines
 	for _, socket := range list.Sockets {
-		// http/https app protocol patterns check
-		match := regex.MatchString(socket.Host)
-		if !match {
-			// testing raw host and port (tcp), report only unsuccessful tests; exclusively non-HTTP/S sockets
-			rawErr := netrunner.RawConnect(socket)
-			if rawErr != nil {
-				messengerText += fmt.Sprintln(socket.Host, ":", socket.Port, rawErr.Error())
-				failedCount++
-			}
-			results.Map[socket.ID] = (rawErr == nil)
-			continue
-		}
+		wg.Add(1)
+		go netrunner.TestSocket(socket, socketChan, &wg)
+	}
 
-		ok, responseCode, httpErr := netrunner.CheckSite(socket)
-		if !ok {
+	// iterate again to fetch results from the channel
+	for _, _ = range list.Sockets {
+		result := <-socketChan
```

Comment:
You are iterating twice, assuming that the number of results is equal to the number of invoked tests - but that is not true since any job that encounters a panic or regexp compile error, would fail silently, causing this loop to block indefinitely.
Instead of looping twice - make a slice of channels where each channel receives results from one test, merge (fan-in) these channels into one and have one `for range` loop iterating over the results from that channel.